### PR TITLE
add max date to international serial

### PR DESCRIPTION
**Pull request by danrlu**
See whether this helps at all. I can't quite test it atm TnT

### CODE REVIEW

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py [hunk 1]
```diff
@@ -279,20 +279,36 @@ def apply_filters(config, subsampling, template_args):
         include_arguments_in_filters = True
         lineage_field = "pango_lineage"
 
-    filter_map = {"filter_start_date": "min_date", "filter_end_date": "max_date"}
-    for filter_name, yaml_key in filter_map.items():
-        value = template_args.get(filter_name)
-        if not value:
-            continue  # This filter isn't set, skip it.
```
Comment (danrlu) [inline src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py:286]: I don't fully understand how this `if not value: continue` differs from the below `if pango_lineages:` so I may have made a bad change here

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py [hunk 2]
```diff
@@ -279,20 +279,36 @@ def apply_filters(config, subsampling, template_args):
...
+            ] = f"--{"min_date".replace('_', '-')} {min_date}"  # ex: --max-date 2020-01-01
         else:
-            subsampling["group"][yaml_key.replace("_", "-")] = str(
-                value
+            subsampling["group"]["min_date".replace("_", "-")] = str(
+                min_date
+            )  # ex: max-date: 2020-01-01
+
+    max_date = template_args.get("filter_end_date")
+    if max_date:
```
Comment (danrlu) [inline src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py:293]: this one too

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py [hunk 3]
```diff
@@ -279,20 +279,36 @@ def apply_filters(config, subsampling, template_args):
...
+            ] = f"--{"max_date".replace('_', '-')} {max_date}"  # ex: --max-date 2020-01-01
+        else:
+            subsampling["group"]["max_date".replace("_", "-")] = str(
+                max_date
+            )  # ex: max-date: 2020-01-01
+            subsampling["group"]["max_date".replace("_", "-")] = str(
+                max_date
             )  # ex: max-date: 2020-01-01
 
     pango_lineages = template_args.get("filter_pango_lineages")
```
Comment (danrlu) [inline src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py:311]: We won't support lineage filter for mpox, is that right? Not seeing it at the UI

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py [hunk 4]
```diff
@@ -279,20 +279,36 @@ def apply_filters(config, subsampling, template_args):
...
+    if min_date:
         # Support date expressions like "5 days ago" in our cron schedule.
-        value = dateparser.parse(value).strftime("%Y-%m-%d")
+        min_date = dateparser.parse(min_date).strftime("%Y-%m-%d")
         if include_arguments_in_filters:
             subsampling["group"][
-                yaml_key
-            ] = f"--{yaml_key.replace('_', '-')} {value}"  # ex: --max-date 2020-01-01
+                "min_date"
+            ] = f"--{"min_date".replace('_', '-')} {min_date}"  # ex: --max-date 2020-01-01
```
Comment (danrlu) [inline src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py:289]: lint is not happy here. Must have done something wildly wrong......

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py (hunk 3 above)
Comment (jgadling) [reply src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py:311]: @ehoops-cz I think we can support this on the backend, but I'm not sure if the designs included this?

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py (hunk 4 above)
Comment (jgadling) [inline src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py]:
```suggestion
            ] = f"--min-date {min_date}"  # ex: --max-date 2020-01-01
```

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py [hunk 5]
```diff
@@ -279,20 +279,36 @@ def apply_filters(config, subsampling, template_args):
...
         else:
-            subsampling["group"][yaml_key.replace("_", "-")] = str(
-                value
+            subsampling["group"]["min_date".replace("_", "-")] = str(
```
Comment (jgadling) [inline src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py]:
```suggestion
            subsampling["group"]["min-date")] = str(
```

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py [hunk 6]
```diff
@@ -279,20 +279,36 @@ def apply_filters(config, subsampling, template_args):
...
+            )  # ex: max-date: 2020-01-01
+
+    max_date = template_args.get("filter_end_date")
+    if max_date:
+        # Support date expressions like "5 days ago" in our cron schedule.
+        max_date = dateparser.parse(max_date).strftime("%Y-%m-%d")
+        if include_arguments_in_filters:
+            subsampling["group"][
+                "max_date"
+            ] = f"--{"max_date".replace('_', '-')} {max_date}"  # ex: --max-date 2020-01-01
```
Comment (jgadling) [inline src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py]:
```suggestion
            ] = f"--max-date {max_date}"  # ex: --max-date 2020-01-01
```

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py [hunk 7]
```diff
@@ -279,20 +279,36 @@ def apply_filters(config, subsampling, template_args):
...
+            subsampling["international_serial_sampling"][
+                "max_date"
+            ] = f"--{"max_date".replace('_', '-')} {max_date}"  # ex: --max-date 2020-01-01
```
Comment (jgadling) [inline src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py]:
```suggestion
            ] = f"--max-date {max_date}"  # ex: --max-date 2020-01-01
```

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py [hunk 8]
```diff
@@ -279,20 +279,36 @@ def apply_filters(config, subsampling, template_args):
...
+        else:
+            subsampling["group"]["max_date".replace("_", "-")] = str(
```
Comment (jgadling) [inline src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py]:
```suggestion
            subsampling["group"]["max-date"] = str(
```

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py [hunk 9]
```diff
@@ -279,20 +279,36 @@ def apply_filters(config, subsampling, template_args):
...
+                max_date
+            )  # ex: max-date: 2020-01-01
+            subsampling["group"]["max_date".replace("_", "-")] = str(
```
Comment (jgadling) [inline src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py]:
```suggestion
            subsampling["group"]["max-date"] = str(
```

File: src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py (hunk 3 above)
Comment (danrlu) [reply src/backend/aspen/workflows/nextstrain_run/build_plugins/type_plugins.py:311]: I found Janeece's design doc. It says "keep lineage for HepC" which prob means not for mpox. 
image

**Review by jgadling — APPROVED** [review]

**Pull request merged**